Version update mp5 (#193)
OpenLiberty/guide-rest-intro: README.adoc

> //  Copyright (c) 2017, 2022 IBM Corporation and others.

Reviewer: general comment: although JAX-RS was an abbreviation that was associated with both the Liberty feature and the java spec, restfulWS only seems to apply to the Liberty feature. The [Jakarta RESTful Web Services spec](https://jakarta.ee/specifications/restful-ws/3.0/jakarta-restful-ws-spec-3.0.html) doesn't use the restfulWS  abbreviation at all. So it seems strange to refer to "Creating a restfulWS application". I don't think we'd want to use that abbreviation unless we were referring specifically to enabling/configuring the Liberty restfulWS feature. So you might want to say "Creating a RESTful Web Services application" and similar throughout. Some from dev might be able to clarify further

[1:36](https://ibm-cloud.slack.com/archives/DP92HKKM3/p1644259009495549)
Here's the contextual statement we added to the docs for JAX-RS and restfulWS features:
Jakarta RESTful Web Services, formerly known as JAX-RS, is a [Jakarta EE](https://www.openliberty.io/docs/22.0.0.1/jakarta-ee.html) platform API. In Open Liberty 21.0.0.12 and later, you can use Jakarta [RESTful Web Services 3.0](https://www.openliberty.io/docs/latest/reference/feature/restfulWS.html) functions by enabling the RESTful Web Services 3.0 feature. The JAX-RS 2.0 and 2.1 implementations are available through the Java RESTful Services feature ([jaxrs-2.0](https://www.openliberty.io/docs/latest/reference/feature/jaxrs-2.0.html) and [jaxrs-2.1](https://www.openliberty.io/docs/latest/reference/feature/jaxrs-2.1.html)). The change in feature name and version reflects the change in API package name prefixes from javax.* to jakarta.*, which is common to all Jakarta EE 9.1 features. For more information, see [Jakarta RESTful Web Services 3.0 behavior changes](https://www.openliberty.io/docs/22.0.0.1/reference/jaxrs-dif.html).

== What you'll learn

You will learn how to build and test a simple RESTful service with Jakarta Restful Web Services and JSON-B, which will expose the JVM's system properties. The RESTful service responds to `GET` requests made to the `\http://localhost:9080/LibertyProject/system/properties` URL.

> The service responds to a `GET` request with a JSON representation of the system properties, where each property is a field in a JSON object like this:

Reviewer: ```suggestion
The service responds to a `GET` request with a JSON representation of the system properties, where each property is a field in a JSON object, like this:
```

== What you'll learn

You will learn how to build and test a simple RESTful service with Jakarta EE RESTful Web Services (restfulWS) and JSON-B, which will expose

> the JVM's system properties. The RESTful service will respond to `GET` requests made to the `\http://localhost:9080/LibertyProject/system/properties` URL.

Reviewer: The RESTful service will respond to `GET` requests made to the `\http://localhost:9080/LibertyProject/system/properties` URL.
--->
The RESTful service responds to `GET` requests made to the `\http://localhost:9080/LibertyProject/system/properties` URL.

}
----

> The design of an HTTP API is essential when creating a web application. The REST API has become the go-to architectural style for building an HTTP API. The Jakarta Restful Web Services API offers functionality for creating, reading, updating, and deleting exposed resources. The Jakarta Restful Web Services API supports the creation of RESTful web services that come with desirable properties, such as performance, scalability, and modifiability.

Reviewer: ```suggestion
The design of an HTTP API is an essential part of creating a web application. The REST API is the go-to architectural style for building an HTTP API. The Jakarta Restful Web Services API offers functions to create, read, update, and delete exposed resources. The Jakarta Restful Web Services API supports the creation of RESTful web services that are performant, scalable, and modifiable.
```

include::{common-includes}/twyb-end.adoc[]

// =================================================================================================
// Creating a RESTful application
// =================================================================================================

== Creating a RESTful application

Navigate to the `start` directory to begin.
// cloud-hosted guide instructions:
ifdef::cloud-hosted[]
```bash
cd /home/project/guide-rest-intro/start
```
endif::[]

[role='command']
include::{common-includes}/devmode-lmp33-start.adoc[]

> The Jakarta Restful Web Services has two key concepts for creating REST APIs. The most obvious one is the resource itself, which is modelled as a class. The second is a RESTful application, which groups all exposed resources under a common path. You can think of the RESTful application as a wrapper for all of your resources.

Reviewer: ```suggestion
Jakarta Restful Web Services defines two key concepts for creating REST APIs. The most obvious one is the resource itself, which is modelled as a class. The second is a RESTful application, which groups all exposed resources under a common path. You can think of the RESTful application as a wrapper for all of your resources.
```

`src/main/java/io/openliberty/guides/rest/SystemApplication.java`
----

SystemApplication.java
[source, Java, linenums, role='code_column hide_tags=copyright']
----
include::finish/src/main/java/io/openliberty/guides/rest/SystemApplication.java[]
----

The [hotspot=systemApplication]`SystemApplication` class extends the `Application` class, which associates all RESTful resource classes in the WAR file with this RESTful application. These resources become available under the common path that's specified with the [hotspot=applicationPath]`@ApplicationPath` annotation. The `@ApplicationPath` annotation has a value that indicates the path in the WAR file that the RESTful application accepts requests from.

// =================================================================================================
// Creating the RESTful resource
// =================================================================================================

== Creating the RESTful resource

> In RESTful application, a single class should represent a single resource, or a group of resources of the same type. In this application, a resource might be a system property, or a set of system properties. It is easy to have a single class handle multiple different resources, but keeping a clean separation between types of resources helps with maintainability in the long run.

Reviewer: ```suggestion
In a RESTful application, a single class represents a single resource, or a group of resources of the same type. In this application, a resource might be a system property, or a set of system properties.A single class can easily handle multiple different resources, but keeping a clean separation between types of resources helps with maintainability in the long run.
```

include::finish/src/main/java/io/openliberty/guides/rest/SystemApplication.java[]
----

The [hotspot=path file=0]`@Path` annotation on the class indicates that this resource responds to the `properties` path in the RESTful Web Services application. The [hotspot=applicationPath file=1]`@ApplicationPath` annotation in the [hotspot=systemApplication file=1]`SystemApplication` class together with the [hotspot=path file=0]`@Path` annotation in this class indicates that the resource is available at the `system/properties` path.

Jakarta Restful Web Services maps the HTTP methods on the URL to the methods of the class by using annotations. Your application uses the [hotspot=get]`GET` annotation to map an HTTP `GET` request to the `system/properties` path.

> The [hotspot=get file=0]`@GET` annotation on the method indicates that this method is to be called for the HTTP `GET` method. The [hotspot=produces file=0]`@Produces` annotation indicates the format of the content that will be returned. The value of the [hotspot=produces file=0]`@Produces` annotation will be specified in the HTTP `Content-Type` response header. For this application, a JSON structure is to be returned. The desired `Content-Type` for a JSON response is `application/json` with `MediaType.APPLICATION_JSON` instead of the `String` content type. Using a constant such as `MediaType.APPLICATION_JSON` is better because if there's a spelling error, a compile failure occurs.

Reviewer: ```suggestion
The [hotspot=get file=0]`@GET` annotation on the method indicates that this method is called for the HTTP `GET` method. The [hotspot=produces file=0]`@Produces` annotation indicates the format of the content that is returned. The value of the [hotspot=produces file=0]`@Produces` annotation is specified in the HTTP `Content-Type` response header. This application returns a JSON structured. The desired `Content-Type` for a JSON response is `application/json`, with `MediaType.APPLICATION_JSON` instead of the `String` content type. Using a constant such as `MediaType.APPLICATION_JSON` is better because a spelling error results in a compile failure.
```
